feat: add script to build kernel with CMA enable

tengomucho opened the pull request:
Added script that builds kernel with CMA enabled. This is required for allocating large memory area for AKD1500.

Commit:
feat: add script to build kernel with CMA enable

Added script that builds kernel with CMA enabled. This is required for
allocating large memory area for AKD1500.

rchevalier-brainchip reviewed (commented):
Will try is before reviewing ;) README could be updated (and the info to revert back to the previously installed kernel too)?

tengomucho commented:
Of course you can test it!
As for the revert to the previous version, it depends on your setup, but there are many sites that explain how to do that, e.g.:

- https://unix.stackexchange.com/questions/432393/downgrade-linux-kernel-without-grub
- https://askubuntu.com/questions/1404722/downgrade-kernel-for-ubuntu-22-04-lts

Do you want me add a mention on how to do that at the end of the README?

rchevalier-brainchip commented:
> Of course you can test it! As for the revert to the previous version, it depends on your setup, but there are many sites that explain how to do that, e.g.:
> 
> * https://unix.stackexchange.com/questions/432393/downgrade-linux-kernel-without-grub
> * https://askubuntu.com/questions/1404722/downgrade-kernel-for-ubuntu-22-04-lts
> 
> Do you want me add a mention on how to do that at the end of the README?

Yes just like done for the installer, a few words would be nice.

Commit:
doc: add info on how to build kernel with CMA in README

rchevalier-brainchip commented on README.md:
```diff
@@ -24,6 +24,35 @@ file after installation and remove `akida_pcie` line.
+## Enable CMA in the kernel
+
+Some systems, e.g.: Ubuntu on x86_64, do not come with CMA (contiguous memory
+allocation) support enabled in the kernel. This is required in AKD1500 if you
```
"in AKD1500" -> "to use AKD1500 devices through PCIe" (?)

rchevalier-brainchip commented on README.md:
````diff
@@ -24,6 +24,35 @@ file after installation and remove `akida_pcie` line.
+./build_kernel_w_cma.sh
+```
+
+The script will create the packages and tell you how to build them and boot
````
tells (?)
them (?)

tengomucho replied:
reworded

hcodina commented on build_kernel_w_cma.sh:
```diff
@@ -0,0 +1,70 @@
+echo
+echo
+
+make -j deb-pkg LOCALVERSION=-cma-akd1500
```
make -j means "no limit" for jobs.
This can lead to bad surprises on a quite loaded system.

rchevalier-brainchip replied:
w/o any -j option script has been running for more than 2.5h on my i7-4770K CPU @3.50GHz (8 cores, 16GB memory) and is not done yet... may be something to be done checinh how many cores avaimable to allwo some dispathing ? (for ex: using ` nproc --all`)

hcodina commented on build_kernel_w_cma.sh:
```diff
@@ -0,0 +1,70 @@
+cp /boot/config-$(uname -r) ./.config
+
+# Modify configuration to enable CMA
+MBYTES_CMA=32
```
Maybe some system have several Akida board plugged in
In that case, 32M may be not enough

tengomucho replied:
we do not use often more than one Akida device, but I will increase it to 64M!

hcodina reviewed (commented):
Seems good.
Several comments in the code itself

hcodina commented on build_kernel_w_cma.sh:
```diff
@@ -0,0 +1,70 @@
+echo "CONFIG_CMA=y" >> ./.config
+echo "CONFIG_CMA_SIZE_MBYTES=$MBYTES_CMA" >> ./.config
+echo "CONFIG_CMA_SIZE_SEL_MBYTES=y" >> ./.config
+
```
In the future, Ubuntu kernel may have these configuration items already set.
Maybe be can check for their value before setting them.

hcodina commented on README.md:
```diff
@@ -24,6 +24,35 @@ file after installation and remove `akida_pcie` line.
+want to use larger amounts of memory to program big models or make full usage of
+the pipeline. To build the kernel packages that include such support you can
+run the dedicated script:
+
```
Mention that this script is for Ubuntu distro.
On other distros, it may not work.

tengomucho commented:
force-pushed after taking into account all comments.

Commit:
fix: corrected cma script exiting immediately

- script was failing when looking if the CMA was already available,
- updated comment to say it will reserve 64M.

rchevalier-brainchip commented on README.md:
```diff
@@ -24,6 +24,36 @@ file after installation and remove `akida_pcie` line.
+etc.
+
+Source: https://unix.stackexchange.com/questions/432393/downgrade-linux-kernel-without-grub
+
```
We could enforce the massage "After a kernel update you need to run the install script again" here again :)

Commit:
review: update CMA script and README

- remove ls in the script to build the kernel, as it seem not to work
  properly (a sync issue). Since that is just to log info, the message
  is removed.
- Added a mention to the need of rebuilding the module after rebooting
  to the new kernel.
- In the README, add a note to tell how to set the MAKEFLAGS env var to
  allow parallel build.

tengomucho commented:
@rchevalier-brainchip can you please review the latest commit?
